### ISSUE TITLE
bug: vue, back button not finding correct route info

### ISSUE THREAD

**Issue by GC-Mark** [ionic-team/ionic-framework#23987]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

The back button stops working under certain scenarios

### Expected Behavior

The back button should work

### Steps to Reproduce

- Clone the linked repo
- `ionic serve`
- Click `Topics` > `Topic 1` > `Article 1`
- Click the back button (_**not**_ the browser back button) to get back to `Topics`
- Click `Topics` > `Topic 2` > `Article 1` or `Article 2`
- The back button will no longer work

### Code Reproduction URL

https://github.com/GC-Mark/testing-app

### Ionic Info

```
Ionic:

   Ionic CLI       : 6.17.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 5.8.1

Capacitor:

   Capacitor CLI      : 3.2.3
   @capacitor/android : not installed
   @capacitor/core    : 3.2.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.1

System:

   NodeJS : v15.14.0 (/usr/local/bin/node)
   npm    : 7.24.1
   OS     : macOS Big Sur
```

### Additional Information

https://user-images.githubusercontent.com/1477806/134936973-44a7acf0-4fdb-4e6e-b96d-04119329b967.mov

**Comment by liamdebeasi** [comment]
Thanks for the issue. I can reproduce this behavior. As a temporary workaround, you can use Ionic Vue v5.6.13.

**Comment by GC-Mark** [comment]
Thanks, `v5.6.13` fixes this for me for now 👍

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/24060, and a fix will be available in an upcoming release of Ionic Framework.

Feel free to keep testing that dev build and let me know if you run into any issues. Thanks!

**Comment by tigohenryschultz** [comment]
I don't think this was fully fixed in 5.8.5. I still encounter it.
